src/vcpkg/commands.usage.cpp

```diff
@@ -0,0 +1,276 @@
+            return Input::check_and_get_full_package_spec(
+                std::string(arg), default_triplet, COMMAND_STRUCTURE.example_text);
+        });
+
+        for (const auto& spec : specs)
+        {
+            Input::check_triplet(spec.package_spec.triplet(), paths);
+        }
+
+        StatusParagraphs status_db = database_load_check(paths);
+
+        if (args.json.value_or(false))
+        {
+            Json::Array json_to_print;
+            for (const auto& spec : specs)
+            {
+                Json::Object& spec_object = json_to_print.push_back(Json::Object{});
+                spec_object.insert("name", Json::Value::string(spec.package_spec.name()));
+                spec_object.insert("triplet", Json::Value::string(spec.package_spec.triplet().to_string()));
+
+                auto it = status_db.find_installed(spec.package_spec);
+                if (it == status_db.end())
+                {
+                    spec_object.insert("installed", Json::Value::boolean(false));
+                    continue;
+                }
+                spec_object.insert("installed", Json::Value::boolean(true));
+                spec_object.insert("usage", to_json(get_cmake_usage((*it)->package, paths)));
+            }
+            System::print2(Json::stringify(json_to_print, {}));
+        }
+        else
+        {
+            for (const auto& spec : specs)
+            {
+                auto it = status_db.find_installed(spec.package_spec);
+                if (it == status_db.end())
+                {
+                    System::printf(System::Color::warning,
+                                   "warning: the package %s:%s is not installed; not printing usage.",
+                                   spec.package_spec.name(),
+                                   spec.package_spec.triplet());
+                    continue;
+                }
+                auto usage = to_string(get_cmake_usage((*it)->package, paths));
+                if (!usage.empty())
+                {
+                    System::print2(usage);
+                }
+                else
+                {
+                    System::printf("The package %s:%s does not have CMake usage instructions.\n",
+                                   spec.package_spec.name(),
+                                   spec.package_spec.triplet());
+                }
+            }
+        }
+    }
+
+    Json::Value to_json(const CMakeUsageInfo& cmui)
+    {
+        auto result = Json::Object{};
+        if (const std::string* usage_file = cmui.usage_file.get())
+        {
+            result.insert("usage-file", Json::Value::string(*usage_file));
+        }
+        else if (const std::string* header_path = cmui.header_to_find.get())
+        {
+            result.insert("header-to-find", Json::Value::string(*header_path));
+        }
+        else if (!cmui.cmake_targets_map.empty())
```

Comment:
negated if followed by else pet peeve

Reply:
Ditto below

Reply:
I disagree with this, since what this is actually checking is `cmui.cmake_targets_map.has_value()` or w/e

Reply:
It still forces a reader to negate a negation in order to figure out what the invariants are in the else.